omnigibson/systems/macro_particle_system.py

```diff
@@ -34,37 +38,38 @@ class MacroParticleSystem(BaseSystem):
     Global system for modeling "macro" level particles, e.g.: dirt, dust, etc.
     """
 
-    # Template object to use -- class particle objet is assumed to be the first and only visual mesh belonging to the
-    # root link of this template object, which symbolizes a single particle, and will be duplicated to generate the
-    # particle system. Note that this object is NOT part of the actual particle system itself!
-    _particle_template = None
+    def __init__(self, name, **kwargs):
+        # Template object to use -- class particle objet is assumed to be the first and only visual mesh belonging to the
+        # root link of this template object, which symbolizes a single particle, and will be duplicated to generate the
+        # particle system. Note that this object is NOT part of the actual particle system itself!
+        self._particle_template = None
 
-    # dict, array of particle objects, mapped by their prim names
-    particles = None
+        # dict, array of particle objects, mapped by their prim names
+        self.particles = dict()
```

Comment:
nit: would be good keep vars that are explicitly initialized later to be None here so that it's more interpretable if there's a bug